common/datavalues/src/data_array_logic.rs

```diff
@@ -47,13 +47,22 @@ impl DataArrayLogic {
                     common_arrow::arrow::compute::not(arr).map_err(ErrorCodes::from)?
                 ))
             }
-            DataColumnarValue::Constant(_v, _size) => {
-                let vec = val.to_array()?;
-                let arr = downcast_array!(vec, BooleanArray)?;
-                let res =
-                    Arc::new(common_arrow::arrow::compute::not(arr).map_err(ErrorCodes::from)?);
-                let vo = DataValue::Boolean(Some(res.value(0)));
-                DataValue::try_into_data_array(&[vo])
+            DataColumnarValue::Constant(v, size) => match v {
```

Comment:
why not just use `vi != 0`

Ok(DataColumnarValue::Constant(vi != 0, *size).to_array()?)

Reply:
Maybe better to add other primitive types support

Reply:
Fix. Please take a look again.